packages/keyring-eth-hd/src/index.js

```diff
@@ -32,14 +31,13 @@ const type = 'HD Key Tree';
 
 class HdKeyring {
   /* PUBLIC METHODS */
-  constructor(opts = {}) {
+  constructor() {
     this.type = type;
     this._wallets = [];
-    this.deserialize(opts);
```

Comment:
This is purely cosmetic, but WDYT of adding some `static` helpers, to avoid having to explicitly call the `deserialize` method (after calling `new`)?

Something like this:
```ts
const keyring = await HdKeyring.fromOptions(opts);
```

This makes it slightly more natural IMO and also makes sure that `deserialize` is being called underneath.

And the implementation would be as simple as:
```ts
static async fromOptions(opts = {}) {
  const keyring = new HdKeyring();
  await keyring.deserialize(opts);
  return keyring;
}
```

I think this way, we still keep somewhat a "sort of" `async` constructor?

Reply:
Yes, I do think it's mostly on the tests, so it's probably not really that useful.

So let's ignore this for now. 🙈